clients/rospy/src/rospy/timer.py

```diff
@@ -204,7 +207,14 @@ def shutdown(self):
         Stop firing callbacks.
         """
         self._shutdown = True
-        
+        self._period = self.__period
+
+    def startup(self):
```

Comment:
What's the protocol for suddenly realizing I had been misinterpreting something rather key to the functionality of a Thread object, and now feel like an idiot?

I was mistaking the run method for **not** exiting when `_shutdown` was flipped to `True`, hence the idea of making it possible to continue its existence after telling it to stop. I suppose it would be possible to have an implementation that did that, but as-is, the code in this PR does not make that functionally possible..

Sorry bout that! Feel free to close this with a vengeance.